client/components/theme-tier/use-theme-tier.js

```diff
@@ -1,5 +1,11 @@
+import { createSelector } from '@automattic/state-utils';
 import getThemeTier from 'calypso/components/theme-tier/get-theme-tier';
 import { useSelector } from 'calypso/state';
 
 export default ( siteId, themeId ) =>
-	useSelector( ( state ) => getThemeTier( state, siteId, themeId ) );
+	useSelector(
+		createSelector(
+			( state ) => getThemeTier( state, siteId, themeId ),
+			( state ) => state.themes.queries[ siteId ]
+		)
+	);
```

Comment:
This change is fine in the wider context of this PR, but please note this cleanup issue I've opened about these hook and selector: https://github.com/Automattic/wp-calypso/issues/85899

Reply:
I'm actively working on that issue.